src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PENamedTypeSymbol.cs

```diff
@@ -1913,15 +1915,39 @@ private MethodDefinitionHandle TryGetExtensionMarkerMethod()
 
             try
             {
-                // They must not contain any instance state
+                // The only expected instance state is the underlying instance field
+                FieldDefinitionHandle foundUnderlyingInstanceField = default;
                 foreach (var field in module.GetFieldsOfTypeOrThrow(_handle))
                 {
-                    if ((module.GetFieldDefFlagsOrThrow(field) & FieldAttributes.Static) == 0)
+                    if (module.GetFieldDefNameOrThrow(field) == WellKnownMemberNames.ExtensionFieldName)
+                    {
+                        if ((module.GetFieldDefFlagsOrThrow(field) & FieldAttributes.Static) != 0)
+                        {
+                            // It must be an instance field
+                            return default;
+                        }
+
+                        if (this.IsStatic)
+                        {
+                            // It's only allowed in non-static extension types
+                            return default;
+                        }
+
+                        Debug.Assert(foundUnderlyingInstanceField.IsNil);
```

Comment:
>Debug.Assert(foundUnderlyingInstanceField.IsNil); [](http://example.com/codeflow?start=24&length=49)

Consider flagging the duplicate as an unsupported case instead of simply asserting.